### PR TITLE
[NFC] Function extraction and renaming of variable for shorter refere…

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
This is a function extraction towards CRM-19273. I haven't changed the logic but it is apparent that
we are changing the contribution financial type to 'the financial type of the last line
item' - which is a little dubious although it might work in practice. We should expose the field &
let people choose it

Before
----------------------------------------
code is in main function

After
----------------------------------------
code is extracted

Technical Details
----------------------------------------
This is  breakdown of changes in #10962

### CODE REVIEW

File: CRM/Price/BAO/LineItem.php [hunk 1]
```diff
@@ -914,19 +914,21 @@ protected function getLineItemsToAlter($submittedLineItems, $entityID, $entity)
...
         else {
-          $lineItemsToAdd[$previousLineItem['price_field_value_id']]['skip'] = TRUE;
+          $submittedLineItem['skip'] = TRUE;
         }
         // for updating the line items i.e. use-case - once deselect-option selecting again
-        if (($previousLineItem['line_total'] != $submittedLineItems[$previousLineItem['price_field_value_id']]['line_total']) ||
-          ($submittedLineItems[$previousLineItem['price_field_value_id']]['line_total'] == 0 && $submittedLineItems[$previousLineItem['price_field_value_id']]['qty'] == 1) ||
-          ($previousLineItem['qty'] != $submittedLineItems[$previousLineItem['price_field_value_id']]['qty'])
+        if (($previousLineItem['line_total'] != $submittedLineItem['line_total'])
+          ||
```
Comment (seamuslee001) [inline CRM/Price/BAO/LineItem.php]: should this go on the previous line?

Comment (eileenmcnaughton) [reply CRM/Price/BAO/LineItem.php]: not sure - at some point I liked it better on the next line - it was a deliberate change - but may I was just in a 'next liney mood'

Comment (mlutfy) [reply CRM/Price/BAO/LineItem.php]: Minor aesthetic, but seems kind of weird to me to have the `||` on a line by itself :

capture d ecran de 2017-11-15 10-46-34

Comment (eileenmcnaughton) [reply CRM/Price/BAO/LineItem.php]: I don't know if it 'looks' right - but the change I just pushed is the most readable to me - ie. having the inners of the OR's separated out

**Pull request merged**